### PR TITLE
Fix map_or and map_or_else. Wrong return. Fix test

### PR DESCRIPTION
**Pull request by check69**
Fixing map_or and map_or_else functions

### CODE REVIEW

File: rusty_results/prelude.py [hunk 1]
```diff
@@ -87,7 +87,7 @@ def map(self, f: Callable[[T], U]) -> "Option[U]":
         ...
 
     @abstractmethod
-    def map_or(self, default: U, f: Callable[[T], U]) -> "Option[U]":
+    def map_or(self, default: U, f: Callable[[T], U]) -> "U":
```
Comment (danielSanchezQ) [inline rusty_results/prelude.py]: No need to enclose `U` with `"`:
just `... -> U`

File: rusty_results/prelude.py [hunk 2]
```diff
@@ -101,7 +101,7 @@ def map_or(self, default: U, f: Callable[[T], U]) -> "Option[U]":
         ...
 
     @abstractmethod
-    def map_or_else(self, default: Callable[[], U], f: Callable[[T], U]) -> "Option[U]":
+    def map_or_else(self, default: Callable[[], U], f: Callable[[T], U]) -> "U":
```
Comment (danielSanchezQ) [inline rusty_results/prelude.py]: Likewise here

**Review by danielSanchezQ — CHANGES_REQUESTED** [review]
Just use `U` no need for the extra `"_"`.
Check all changes, not just the commented ones.

File: rusty_results/prelude.py (hunk 1 above)
Comment (check69) [reply rusty_results/prelude.py]: Done

File: rusty_results/prelude.py (hunk 2 above)
Comment (check69) [reply rusty_results/prelude.py]: Done

**Comment by danielSanchezQ** [comment]
Awesome, thanks!

**Pull request merged**